iac/main/resources/sustainability.yml

```diff
@@ -101,3 +101,44 @@ SustainabilityBucketRole:
           Principal:
             Service:
               - s3.amazonaws.com
+
+SustainabilityCrawlerRole:
+  Type: AWS::IAM::Role
+  Condition: IsADMEnvironment
+  Properties:
+    RoleName: !Sub ${Environment}-sustainability-glue-crawler-role
+    AssumeRolePolicyDocument:
+      Version: '2012-10-17'
+      Statement:
+        - Effect: Allow
+          Principal:
+            Service: [glue.amazonaws.com]
+          Action: ['sts:AssumeRole']
+    ManagedPolicyArns:
+      - arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole
+    Path: /
+    Policies:
+      - PolicyName: !Sub ${Environment}-sustainability-glue-crawler-policy
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+            - Effect: Allow
+              Action:
+                - 'glue:GetConnection'
+                - 'glue:GetCrawler'
+                - 'glue:CreateTable'
+                - 'glue:UpdateCrawler'
+                - 'glue:CreatePartition'
+                - 'glue:BatchCreatePartition'
+              Resource: '*'
+            - Effect: Allow
+              Action:
+                - 's3:GetObject'
+                - 's3:ListObject'
+              Resource:
+                # Cost Usage Report buckets in SRE account
+                - 'arn:aws:s3:::cid-921370741319-shared'
```

Comment:
not a huge fan of having this in a public repo event though they're technically not a secret: https://docs.aws.amazon.com/accounts/latest/reference/manage-acct-identifiers.html

ill approve anyway